File: 0121-best-time-to-buy-and-sell-stock/solution.java
class Solution {
    public int maxProfit(int[] p) {
        int min=p[0],max=0;
        for(int i=0;i<p.length;i++){
            min=Math.min(min,p[i]);
            max=Math.max(max,p[i]-min);
        }
        return max;
         
    }
}

File: 0121-best-time-to-buy-and-sell-stock/solution.cpp
class Solution {
public:
    int maxProfit(vector<int>& p) {
        int mini=p[0],maxi=0;
        for(int i=0;i<p.size();i++){
            mini=min(mini,p[i]);
            maxi=max(maxi,p[i]-mini);
        }
        return maxi;
    }
};
